platform/kube/controller.go

```diff
@@ -568,6 +569,38 @@ func (c *Controller) HostInstances(addrs map[string]bool) []*model.ServiceInstan
 	return out
 }
 
+// TODO error indicator
+func (c *Controller) GetIstioServiceAccounts(hostname string) []string {
+	saArray := []string{}
+	saSet := make(map[string]bool)
+	name, namespace, err := parseHostname(hostname)
+	if err != nil {
+		glog.V(2).Infof("parseHostname(%s) => error %v", hostname, err)
+		return saArray
+	}
+	svc, exists := c.serviceByKey(name, namespace)
+	if !exists {
+		glog.Errorf("Failed to get service for hostname %s.", hostname)
+		return saArray
+	}
+	lo := v1.ListOptions{
+		LabelSelector: labels.Set(svc.Spec.Selector).String(),
+	}
+	pods, err := c.client.client.Pods(svc.Namespace).List(lo)
+	if err != nil {
+		glog.Errorf("Failed to get pods for service %s.", hostname)
+		return saArray
+	}
+	for _, p := range pods.Items {
+		sa := "istio:" + p.Spec.ServiceAccountName + "." + namespace
```

Comment:
We also need `domain` per the design doc. For in-cluster only identity, we can use `cluster.local` as the domain